### DISCUSSION TITLE
Idea: Internected Search

### DISCUSSION THREAD

**Discussion by goeo-** [bluesky-social/atproto#2352] [Bluesky Lexicons]
What if app.bsky.feed.searchPosts worked similarly to app.bsky.feed.getFeed and for example, feeds could implement their own search? Different searches could use different ways of matching the query to posts, too.

**Comment by bnewbold** [comment]
Two ways of doing this come to mind:

- first would be making it work like feed gens, which would mean sending a request to the AppView with a "search service" indicated. the appview would query that service for a "skeleton" (result set of just URIs), then hydrate the responses. This would require new concepts, new service type, etc.
- second would be to have the client drive this, by setting the `atproto-proxy` header to indicate a search service with a request to `app.bsky.unspecced.searchPostsSkeleton`. this would return just a skeleton, which the client would then send to a full AppView to `app.bsky.feed.getPosts`, which would "hydrate" the skeleton. this involves an "unspecced" endpoint, and there might be some ergonomics around result set size (`getPosts` would remove or tombstone any muted, blocked, deleted, etc posts, resulting in smaller set size), but I believe it all works today.

**Reply by goeo-** [reply]
i think the former is a better solution here, the latter feels like a hack and i think the internected search should be the same interface eventually as the appview based search
